Project Cleaning and Restructuring

NPN opened the pull request:
To prepare the project for next year's website, I've cleaned and restructured it.

All files from 2016 and Meteor files have been removed.

Restructure changes:
- Project is now structured around Mustache templates (with their corresponding YAML files), compiled with the Mustache ruby gem. These templates are then manually concatenated into one HTML file by the Makefile, which also specifies the concatenation order. The `head` template is special, and is put in the head, obviously. Minification/beautification could be added in too.
- JS files are in their own directory. Again, they are manually concatenated to create a "bundle." Uglification could be added here, as well as wrapping the whole thing in a `$(document).ready()`. External JS dependencies could also be handled better, as now they must be put in the `head` template.
- Added Compass and fiddled with `config.rb` to produce compressed css.
- Sass is structured as many partials and one main file which includes said partials to create a css "bundle." I'm assuming here that you have one file per template, but that doesn't have to be true.
- A custom Makefile compiles everything.
- As we seem to be leaning towards ruby tools here, I added a Gemfile/.lock to track gem dependencies.
- Added CNAME (previously missing in develop repo)
- Updated .gitignore
- Removed Google Analytics (since we weren't using it)

NPN committed:
Remove old Meteor files

NPN committed:
Remove all 2016 files and reorganize

NPN committed:
Restructure project

- Structure project to use Mustache templates
- Move JS files to separate directory
- Add Compass, structure Sass as many partials and one main file
- Write custom Makefile, which does basic concatenation of JS and HTML
- Add Gemfile to track gem dependencies
- Add CNAME (previously missing in develop repo)
- Update .gitignore

Danappelxx commented:
- `.meteor` directory is still there
- Can we include a `make install` step that gets all dependencies?

NPN committed:
Remove .meteor

NPN committed:
Add hacky HTML indentation

NPN committed:
Clean up HTML

NPN committed:
Change CSS dir to build dir

NPN committed:
Add 2016 website

NPN committed:
Remove extra space added from hacky indentation

NPN committed:
Only indent non-empty lines

NPN committed:
Use simply expanded variables and parentheses

NPN committed:
Update README

NPN committed:
Add production target to copy build files

NPN committed:
Remove a useless use of cat

NPN committed:
Add make watch via fswatch and live.js

Mustache -e flag removed because "file not found errors" kept happening
when using make watch. The js bundle is also now wrapped in
$(document).ready, creating a jQuery dependency.

NPN committed:
Fix Vim swap file regex

NPN committed:
Add WATCH_DIRS variable
